plugins/modules/azure_rm_securitygroup.py

```diff
@@ -475,6 +475,8 @@ def compare_rules_change(old_list, new_list, purge_list):
 
 
 def compare_rules(old_rule, rule):
+    def compare_list_rule(old_rule, rule, key):
```

Comment:
It would be even better if the compare_list_rule function were defined outside compare_rules. Thank you!

Reply:
I was following the similar example here: https://github.com/ansible-collections/azure/blob/5073ac9ad051647bda66b957fabd3b610e1def73/plugins/modules/azure_rm_securitygroup.py#L439. I can change, but guess it should be a static method as it doesn't need anything from the class (unless you're ignoring linter warnings about static methods?)

Reply:
yes， I will confirm with developer! Thank you very much!